### PR TITLE
Feature/at 7367

### PR DESCRIPTION
**Pull request by robertmccardell-ccpo**
ATAT Upload functionality

### CODE REVIEW

File: src/components/ATATFileListItem.vue [hunk 1]
```diff
@@ -42,20 +52,30 @@
             role="button"
             :id="'FileLink0' + index"
             target="_blank"
-            class="_text-link d-flex align-center justify-start"
+            :href="uploadingFileObj.link"
+            :class="[
+              uploadingFileObj.isErrored ? 'error--text' : '_text-link',
+              ' d-flex align-center justify-start]',
```
Comment (ericyoungquist-ccpo) [inline src/components/ATATFileListItem.vue:58]: remove `]` inside class list

File: src/store/acquisitionPackage/index.ts [hunk 2]
```diff
@@ -1924,13 +1932,105 @@ export class AcquisitionPackageStore extends VuexModule {
...
+  async loadAttachments(key: string): Promise<AttachmentDTO[] | undefined> {
+    const storeData = this as unknown as Record<string, unknown>;
+
+    try {
+      //attachment table data is stored as a comma seperated
+      // string list on the acquisition package object and in the store
+      const tableIdList = storeData[key] as string;
+      const tableIds = tableIdList.length ? tableIdList.split(",") : [];
+
+      if (tableIds.length == 0) {
```
Comment (ericyoungquist-ccpo) [inline src/store/acquisitionPackage/index.ts:1958]: use `===`

File: src/components/ATATFileUpload.vue [hunk 3]
```diff
@@ -182,16 +194,90 @@ export default class ATATFileUpload extends Vue {
...
+        recordId: "",
+        isErrored: false,
+        isUploaded: false,
+      });
+    });
+    this.uploadFiles();
+  }
+
+  private uploadFiles(): void {
+    // this.validFiles.forEach((uploadingFile) => {
```
Comment (ericyoungquist-ccpo) [inline src/components/ATATFileUpload.vue:234]: remove commented line

File: src/components/ATATFileUpload.vue [hunk 4]
```diff
@@ -182,16 +194,90 @@ export default class ATATFileUpload extends Vue {
...
+    for (let i = 0; i < this.validFiles.length; i++) {
+      //wire up file upload here
+      let uploadingFileObj = this.validFiles[i] as uploadingFile;
+
+      // only new files are uploaded
+      if (!uploadingFileObj.isUploaded) {
+        // this.validFiles.push(vFile);
```
Comment (ericyoungquist-ccpo) [inline src/components/ATATFileUpload.vue:241]: remove commented line

File: src/components/ATATFileUpload.vue [hunk 5]
```diff
@@ -182,16 +194,90 @@ export default class ATATFileUpload extends Vue {
...
+              //table_sys_id the unique id of the table/record
+
+              const { download_link, sys_id, table_sys_id } = result.attachment;
+              uploadingFileObj.link = download_link || "";
+              uploadingFileObj.attachmentId = sys_id || "";
+              uploadingFileObj.recordId = table_sys_id;
+              uploadingFileObj.isUploaded = true;
+            })
+            .catch((error) => {
+              // uploadingFile.isErrored === error.
```
Comment (ericyoungquist-ccpo) [inline src/components/ATATFileUpload.vue:268]: remove commented line

File: src/components/ATATFileUpload.vue [hunk 6]
```diff
@@ -182,16 +194,90 @@ export default class ATATFileUpload extends Vue {
...
+              //file upload error occurred
+
+              uploadingFileObj.isErrored = true;
+              console.log(`file upload error ${error}`);
+            });
+        }, i * 1000);
+
+        // });
```
Comment (ericyoungquist-ccpo) [inline src/components/ATATFileUpload.vue:276]: remove commented line

File: src/store/acquisitionPackage/index.ts [hunk 7]
```diff
@@ -1924,13 +1932,105 @@ export class AcquisitionPackageStore extends VuexModule {
...
+        ...this.acquisitionPackage,
         contract_type: sys_id,
       } as AcquisitionPackageDTO);
     } catch (error) {
       throw new Error(`error occurred saving PoP data ${error}`);
     }
   }
+
+  /**
+   * A method to retrieve data from tables that are used strictrly for attaching files
```
Comment (ericyoungquist-ccpo) [inline src/store/acquisitionPackage/index.ts:1944]: misspelling in comment `strictrly` but I do like that word :)

File: src/store/acquisitionPackage/index.ts [hunk 8]
```diff
@@ -1924,13 +1932,105 @@ export class AcquisitionPackageStore extends VuexModule {
...
+    recordId,
+  }: {
+    key: string;
+    attachmentId: string;
+    recordId: string;
+  }): Promise<void> {
+    const storeData = this as unknown as Record<string, unknown>;
+
+    try {
+      //attachment table data is stored as a comma seperated
```
Comment (ericyoungquist-ccpo) [inline src/store/acquisitionPackage/index.ts:1992]: `separated` (sp)

File: src/components/ATATFileListItem.vue (hunk 1 above)
Comment (tonysumner-ccpo) [reply src/components/ATATFileListItem.vue:58]: Done

File: src/components/ATATFileUpload.vue (hunk 3 above)
Comment (tonysumner-ccpo) [reply src/components/ATATFileUpload.vue:234]: done

File: src/components/ATATFileUpload.vue (hunk 5 above)
Comment (tonysumner-ccpo) [reply src/components/ATATFileUpload.vue:268]: done

File: src/components/ATATFileUpload.vue (hunk 4 above)
Comment (tonysumner-ccpo) [reply src/components/ATATFileUpload.vue:241]: done

File: src/components/ATATFileUpload.vue (hunk 6 above)
Comment (tonysumner-ccpo) [reply src/components/ATATFileUpload.vue:276]: done

File: src/store/acquisitionPackage/index.ts (hunk 2 above)
Comment (tonysumner-ccpo) [reply src/store/acquisitionPackage/index.ts:1958]: done

File: src/store/acquisitionPackage/index.ts (hunk 7 above)
Comment (tonysumner-ccpo) [reply src/store/acquisitionPackage/index.ts:1944]: resolved

File: src/store/acquisitionPackage/index.ts (hunk 8 above)
Comment (tonysumner-ccpo) [reply src/store/acquisitionPackage/index.ts:1992]: Done

**Review by ericyoungquist-ccpo — APPROVED** [review]

**Pull request merged**